.github/workflows/ci-goreleaser.yaml

```diff
@@ -49,7 +49,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: '~1.21'
```

Comment:
While this looks fine and low-risk, should we apply those changes first in the core/contrib repos, to ensure that everything is fully tested before preparing a release?

Like here:

https://github.com/open-telemetry/opentelemetry-collector/blob/03b56b6bca3fc2fca53a19fea849538de4dfda47/.github/workflows/build-and-test.yml#L22

Reply:
Yes, the PRs have already been made to collector core and contrib.

Reply:
My PR is merely forcing a minimum version. minor version `v1.21` is already being used in these workflows.